cms/serializers.py

```diff
@@ -48,6 +49,13 @@ class ProgramPageSerializer(serializers.ModelSerializer):
     slug = serializers.SerializerMethodField()
     faculty = FacultySerializer(source='faculty_members', many=True)
     courses = CourseSerializer(source='program.course_set', many=True)
+    electives_required_number = serializers.SerializerMethodField()
+
+    def get_electives_required_number(self, programpage):
+        """Get number of required elective courses for a program"""
+        if programpage.program and programpage.program.electives_set.exists():
+            return programpage.program.electives_set.all().aggregate(Sum('required_number'))['required_number__sum']
+        return 0
```

Comment:
@annagav  Actually, I did test it again with multiple elective sets and it works for me. Since programs are associated with elective sets using a foreign key, the part where ```program.electives_set``` is used iterates with a reverse foreign key and it contains all the elective sets associated with a program.

**Scenario I tested:**
1. Created an elective set with required number value 1.
2. Created another elective set with required number value 1.
3. Associated both of the sets with a single/same program.

The above query returns the right value summing up the required number field from all the associated sets. e.g. 2 in the above scenario. I have also tested with other scenarios changing the required number value and it seems to work.

Can you please give it a try as well, perhaps this might work for you as well.

Reply:
I don't think that we want to display the sum of all required courses over all sets. 
If there are multiple sets, for each set we should be showing it's required number of courses.
So if you have two elective sets, in the program page it should right:
electives (1 of 2):
course 1
course 2
electives (2 of 3):
course 3
course 4 
course 5

So in other words the two sets should be kept separate from each other.

Reply:
This might be a little different in our case I suppose. Previously, The implementations of the course list used to mark all the elective courses from all the associated elective sets cumulatively. And as per the new design link attached in the associated issue, there is no set wise grouping but course type wise (Core or Elective). 

Maybe we can refer to @pdpinch for this change, we might need the updated design if we are going to group the courses by the elective set as well.

Reply:
Sorry, I don't think I understand. Anna's comment makes sense to me, but I don't know what might have been done previously. 

To go back to what is needed. We have 
1. the DEDP program, which has 3 core courses and then the learner much complete 2 of 3 electives. You can see this on prod now at https://micromasters.mit.edu/dedp/
2. a new program, which will have 4 core courses and then learner must complete 1 of 2 electives. 

We don't have any plans at the moment for having more than 2 sets for a course (1 core and 1 elective), but I suppose it could happen in the future.

Reply:
Thanks for the details. I'll go ahead and add these changes.